snaps/features/data-storage.md

```diff
@@ -0,0 +1,120 @@
+---
+description: Store encrypted and unencrypted data within a Snap.
+sidebar_position: 5
+---
+
+# Data storage
+
+You can store and manage sensitive information within a Snap using encrypted storage, or
+non-sensitive information using unencrypted storage.
+Use the [`snap_manageState`](../reference/snaps-api.md#snap_managestate) API method to persist up to
+100 MB of data to the user's disk and retrieve it at will.
+We recommend using this method for storing data in a Snap long term.
+
+## Steps
+
+### 1. Get permission to store data
+
+Request the [`snap_manageState`](../reference/snaps-api.md#snap_managestate) permission.
+Add the following to your Snap's manifest file:
+
+```json title="snap.manifest.json"
+"initialPermissions": {
+  "snap_manageState": {}
+}
+```
+
+### 2. Use encrypted storage
+
+By default, [`snap_manageState`](../reference/snaps-api.md#snap_managestate) automatically encrypts
+data using a Snap-specific key before storing it on the user's disk, and automatically decrypts it
+when retrieved.
+This is useful to store sensitive information, such as passwords.
+
+The following example uses `snap_manageState` to store some data using the `update` operation, and
+retrieves the data at a later time using the `get` operation.
+Once there's no need to store the data anymore, it clears the Snap's state using the `clear` operation.
```

Comment:
I think refrain from using "once" as a synonym for after 
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/o/once

```suggestion
When the data is no longer required, the Snap's state is cleared using the `clear` operation.
```